lib/libxdp/xsk_def_xdp_prog_5.3.c

```diff
@@ -0,0 +1,39 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+
+#include <linux/bpf.h>
+#include <bpf/bpf_helpers.h>
+#include <xdp/xdp_helpers.h>
+
+#include "xsk_def_xdp_prog.h"
+
+#define DEFAULT_QUEUE_IDS 64
+
+struct {
+        __uint(type, BPF_MAP_TYPE_XSKMAP);
+        __uint(key_size, sizeof(int));
+        __uint(value_size, sizeof(int));
+	__uint(max_entries, DEFAULT_QUEUE_IDS);
+} xsks_map SEC(".maps");
+
+struct {
+        __uint(priority, 20);
+	__uint(XDP_PASS, 1);
+} XDP_RUN_CONFIG(xsk_def_prog);
+
+/* This is the program for 5.3 kernels and older. */
+SEC("xdp/xsk_def_prog")
+int xsk_def_prog(struct xdp_md *ctx)
```

Comment:
Any reason these two progs couldn't just be kept in the same object (but in different sections)?

Reply:
The code in xsk.c turned out simpler if I kept them in two separate files. Actually had them in the same first.

Reply:
Alright, not a huge deal, just thought it might be simpler if we ever needed to update the map definition or something. But that would require other changes as well, so fine with just keeping it like this.